### ISSUE TITLE
[rllib] Access info dict in callback function

### ISSUE THREAD

**Issue by 0luhancheng0** [ray-project/ray#8350]
I was trying to compute the custom metric according to info dict returned along with each transition. 

I'm following the example at 
https://github.com/ray-project/ray/blob/master/rllib/examples/custom_metrics_and_callbacks.py

I was wondering how can I access the info dict returned by each transition within the callback? Thank you

**Comment by 0luhancheng0** [comment]
It seems episode.last_info_for() always return None, not sure if this is expected

**Comment by ericl** [comment]
last_info_for() works for me (w/the default cartpole it returns {}). With a custom env, it returns the info dict as expected.

**Comment by 0luhancheng0** [comment]
Hi @ericl , thank you for replying. 
I'm sorry that I didn't mention i was running my custom MultiAgentEnv. But the same behaviour can also be observed with one of your examples. 

A simple script as following
```
import gym 
import ray
from ray.tune.registry import register_env
from ray.rllib.agents.pg import PGTrainer
from ray.rllib.agents.callbacks import DefaultCallbacks
from ray.rllib.examples.env.multi_agent import BasicMultiAgent


class SampleCallback(DefaultCallbacks):

    def on_episode_step(self, worker, base_env,
                        episode, **kwargs):
        print(f"last info for: {episode.last_info_for()}")
        return

config = {
    'env_config': {
        'num_agents': 3
    }, 
    'callbacks': SampleCallback
}
ray.init(num_cpus=4)
register_env('basic', lambda env_config: BasicMultiAgent(env_config['num_agents']))
trainer = PGTrainer(config=config, env='basic')
trainer.train()

```

A snippet of output: 
```
last info for: None
last info for: None
last info for: None
last info for: None
last info for: None
last info for: None
```

I was expecting last_info_for() to return an empty directory instead of None.

**Comment by krfricke** [comment]
Your code is almost correct, but since you're running a MultiAgentEnv, you will have to specify the agent ID when fetching the last info dict, like so:

```
print(f"last info for: {episode.last_info_for(0)}")
```

You can, of course, loop through all agents to fetch all info dicts:

```
for i in range(3):
    print(f"last info for agent {i}: {episode.last_info_for(i)}")
```
